src/main/java/com/iota/iri/BundleValidator.java

```diff
@@ -67,14 +113,17 @@
                                     curlInstance.absorb(transactionViewModel2.trits(), TransactionViewModel.ESSENCE_TRINARY_OFFSET, TransactionViewModel.ESSENCE_TRINARY_SIZE);
                                 }
                                 curlInstance.squeeze(bundleHashTrits, 0, bundleHashTrits.length);
-                                if (Arrays.equals(instanceTransactionViewModels.get(0).getBundleHash().trits(), bundleHashTrits)) {
-
+                                //verify bundle hash is correct
+                                if (Arrays.equals(instanceTransactionViewModels.get(0).getBundleHash().trits(), bundleHashTrits))  {
+                                    //normalizing bundle in preparation for sig verification
                                     ISSInPlace.normalizedBundle(bundleHashTrits, normalizedBundle);
 
                                     for (int j = 0; j < instanceTransactionViewModels.size(); ) {
 
                                         transactionViewModel = instanceTransactionViewModels.get(j);
-                                        if (transactionViewModel.value() < 0) { // let's recreate the address of the transactionViewModel.
+                                        //if it is a spent transaction that should be signed
```

Comment:
```suggestion
                                        //if it is a spent transaction then it should be signed
```

Reply:
no,
"then it" implies that now we will sign it. We are not signing, just validating a sig.

Reply:
👍